server/src/main/java/keywhiz/ServiceModule.java

```diff
@@ -88,18 +88,15 @@ public ServiceModule(KeywhizConfig config, Environment environment) {
 
   // ManagedDataSource
 
-  /**
-   * TODO: make sure that the read-write health check does not impact availability of
-   * the service.
-   * TODO: make sure that the readonly has a health check and marks the service as unhealthy if
-   * the connection is down. Optionally, fallback to read-write if that connection is healthy.
-   */
   @Provides @Singleton ManagedDataSource dataSource(Environment environment,
       KeywhizConfig config) {
     DataSourceFactory dataSourceFactory = config.getDataSourceFactory();
     ManagedDataSource dataSource = dataSourceFactory.build(environment.metrics(), "postgres-writable");
     environment.lifecycle().manage(dataSource);
 
+    environment.healthChecks().register("db-read-write-health",
+        new KeywhizDbHealthCheck(dataSource, true));
```

Comment:
An inline comment like `true /* only log unhealthy */)` would make this more clear. Either that or switching the boolean to an enum.